### PR TITLE
DAOS-7338 test: Exclude and Reintegration with EC Objects

### PR DESCRIPTION
**Pull request by dinghwah**
Update test.yaml to add with EC oclass for ior.
Skip-unit-tests: true
Allow-unstable-test: true
Test-tag: test_nvme_pool_excluded test_nvme_pool_extend
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by shimizukko — APPROVED** [review]

File: src/tests/ftest/nvme/pool_exclude.yaml [hunk 1]
```diff
@@ -33,36 +33,42 @@ pool:
...
+      dfs_oclass: RP_3G6
+      dfs_dir_oclass: RP_3G6
+    ec_4p2gx:
+      <<: *ior_flags_common
+      dfs_oclass: EC_4P2GX
+      dfs_dir_oclass: EC_4P2GX
+    ec_8p2gx:
+      <<: *ior_flags_common
+      dfs_oclass: EC_8P2GX
+      dfs_dir_oclass: EC_8P2GX
```
Comment (daltonbohning) [inline src/tests/ftest/nvme/pool_exclude.yaml:71]: Just FYI that IOR doesn't use `dfs_dir_oclass`. We set this a lot, and there's a default value of `dfs_dir_oclass=SX`, but IOR only works on files, not directories. Okay to keep this for now - eventually I'd like to clean it all up and remove the unused `dfs_dir_oclass` param from the utils.

File: src/tests/ftest/nvme/pool_exclude.yaml [hunk 2]
```diff
@@ -33,36 +33,42 @@ pool:
...
-test_obj_class:
-  oclass:
-    - RP_3G6
-    - RP_4G1
+  dfs_destroy: False
+  iorflags: !mux
+    rp_3g6:
+      <<: *ior_flags_common
+      dfs_oclass: RP_3G6
+      dfs_dir_oclass: RP_3G6
```
Comment (daltonbohning) [inline src/tests/ftest/nvme/pool_exclude.yaml:63]: Any specific reason to keep these as `G6` instead of `GX` like the other variants below?

**Review by daltonbohning — APPROVED** [review]
Looks like the latest commit only ran the exclude test case - not the extend case:
`Test-tag: test_nvme_pool_excluded`
Was this intentional?

File: src/tests/ftest/nvme/pool_exclude.yaml (hunk 2 above)
Comment (dinghwah) [reply src/tests/ftest/nvme/pool_exclude.yaml:63]: RP_3G6 is the existing oclass in CI.

Comment (daltonbohning) [reply src/tests/ftest/nvme/pool_exclude.yaml:63]: Right, but the new variants use `GX`, so I wonder if this should also use `GX`. (I don't know)

Comment (dinghwah) [reply src/tests/ftest/nvme/pool_exclude.yaml:63]: Added @Ravi and @Phil review if update to GX is needed.

Comment (daltonbohning) [reply src/tests/ftest/nvme/pool_exclude.yaml:63]: Might be a question for a developer. The difference is that `GX` will allow DAOS to calculate the maximum group number based on the data, parity, and number of system targets.

Comment (rpadma2) [reply src/tests/ftest/nvme/pool_exclude.yaml:63]: I think you can use GX... It shouldn't make a difference.

Comment (dinghwah) [reply src/tests/ftest/nvme/pool_exclude.yaml:63]: Added @Samir for review.

Comment (daltonbohning) [reply src/tests/ftest/nvme/pool_exclude.yaml:63]: BTW not a blocker for me, since it's pre-existing

**Pull request merged**